src/utils_request.js

```diff
@@ -0,0 +1,75 @@
+import contentType from 'content-type';
+import * as url from 'url';
+import _ from 'underscore';
+import httpRequest from '@apify/http-request';
+
+export const FIREFOX_MOBILE_USER_AGENT = 'Mozilla/5.0 (Android; Mobile; rv:14.0) Gecko/14.0 Firefox/14.0';
+export const FIREFOX_DESKTOP_USER_AGENT = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:68.0) Gecko/20100101 Firefox/68.0';
+
+export const REQUEST_AS_BROWSER_DEFAULT_OPTIONS = {
+    countryCode: 'US',
+    languageCode: 'en',
+    headers: {},
+    method: 'GET',
+    useMobileVersion: false,
+    useBrotli: true,
+    json: false,
+};
+/**
+ * Sends a HTTP request that looks like a request sent by a web browser,
+ * fully emulating browser's HTTP headers.
+ *
+ * This function is useful for web scraping of websites that send the full HTML in the first response.
+ * Thanks to this function, the target web server has no simple way to find out the request
+ * hasn't been sent by a full web browser. Using a headless browser for such requests
+ * is an order of magnitude more resource-intensive than this function.
+ *
+ * Currently, the function sends requests the same way as Firefox web browser does.
+ * In the future, it might add support for other browsers too.
+ *
+ * Internally, the function uses httpRequest function from the [@apify/httpRequest](https://github.com/apifytech/http-request)
+ * NPM package to perform the request.
+ * All `options` not recognized by this function are passed to it,
+ * so see it for more details.
+ *
+ * @param options.url
+ *  URL of the target endpoint. Supports both HTTP and HTTPS schemes.
+ * @param [options.method=GET]
+ *  HTTP method.
+ * @param [options.headers]
+ *  Additional HTTP headers to add. It's only recommended to use this option,
+ *  with headers that are typically added by websites, such as cookies. Overriding
+ *  default browser headers will remove the masking this function provides.
+ * @param [options.languageCode=en]
+ *  Two-letter ISO 639 language code.
+ * @param [options.countryCode=US]
+ *  Two-letter ISO 3166 country code.
+ * @param [options.isMobile]
+ *  If `true`, the function uses User-Agent of a mobile browser.
+ *
+ * @return {http.IncomingMessage}
+ * @memberOf utils
+ * @name requestAsBrowser
+ */
+export const requestAsBrowser = async (options) => {
+    const opts = _.defaults({}, options, REQUEST_AS_BROWSER_DEFAULT_OPTIONS);
+
+    const parsedUrl = url.parse(opts.url);
+    const abortFunction = (res) => {
```

Comment:
I'd move this to `REQUEST_AS_BROWSER_DEFAULT_OPTIONS` which makes it overridable